packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -3093,7 +3143,7 @@ class _NavigationBarComponentsTransition {
     return PositionedTransition(
       rect: animation.drive(positionTween),
       child: FadeTransition(
-        opacity: fadeInFrom(0.3),
+        opacity: fadeInFrom(0.0),
```

Comment:
Spoke offline. The title was completely fading out too early, but it needs to be partially visible during the whole transition.